exporters/tests/test_in_memory_metric.py

```diff
@@ -0,0 +1,136 @@
+)
+
+metric_labels = list(_pelorus_metric_to_dict(CommitTimePelorusPayload).values())
+
+in_memory_test_committime_metrics = PelorusGaugeMetricFamily(
+    "test_committime_metrics",
+    "Test timestamp",
+    labels=metric_labels,
+)
+
+
+class CustomCommitCollector(Collector):
+    def __init_subclass__(cls) -> None:
+        super().__init_subclass__()
+
+        # make sure __hash__ is something prometheus' registry can handle properly.
+        cls.__hash__ = lambda self: id(self)  # type: ignore
+
+    def collect(self) -> PelorusGaugeMetricFamily:
+        yield in_memory_test_committime_metrics
+
+
+@pytest.mark.parametrize(
+    "name,timestamp,image_hash,namespace,commit_hash",
+    [
+        (
+            "todolist",
+            "timestamp_str",
+            "sha256:af4092ccbfa99a3ec1ea93058fe39b8ddfd8db1c7a18081db397c50a0b8ec77d",
+            "mynamespace",
+            "5379bad65a3f83853a75aabec9e0e43c75fd18fc",
+        ),
+    ],
+)
+def test_pelorus_gauge_metric_family(
+    name, timestamp, image_hash, namespace, commit_hash
+):
+    # custom_collector = load_and_log(CustomCommitCollector)
+    custom_collector = CustomCommitCollector()
+    metric_labels = list(_pelorus_metric_to_dict(CommitTimePelorusPayload).values())
+
+    commit_payload = CommitTimePelorusPayload(
+        app=name,
+        timestamp=timestamp,
+        image_sha=image_hash,
+        namespace=namespace,
+        commit_hash=commit_hash,
+    )
+
+    prometheus_commit_metric = pelorus_metric_to_prometheus(commit_payload)
+    in_memory_test_committime_metrics.add_metric(
+        commit_payload.commit_hash, prometheus_commit_metric, commit_payload.timestamp
+    )
+
+    REGISTRY.register(custom_collector)
+
+    metric_labels = {
+        "app": name,
+        "image_sha": image_hash,
+        "commit_hash": commit_hash,
+        "namespace": namespace,
+    }
+
+    query_result = REGISTRY.get_sample_value(
+        "test_committime_metrics",
+        labels=metric_labels,
+    )
+
+    assert query_result == timestamp
+
+
+def test_all_models_have_prometheus_mappings():
+    import webhook.models.pelorus_webhook as pelorus_webhook
+
+    test_models = []
+
+    for cls in pelorus_webhook.__dict__.values():
+        if isinstance(cls, type) and issubclass(cls, PelorusPayload):
+            test_models.append(cls)
+
+    for test_model in test_models:
+        metric = _pelorus_metric_to_dict(test_model)
+        assert bool(metric)  # dict should not be empty
+
+
+def test_model_do_not_have_prometheus_mapping():
+    class MyPlugin(PelorusPayload):
+        pass
+
+    with pytest.raises(TypeError) as type_error:
+        _pelorus_metric_to_dict(MyPlugin)
+    assert "Improper prometheus data model" in str(type_error.value)
+
+
+@mock.patch(
+    "webhook.store.in_memory_metric._pelorus_metric_to_dict",
+    return_value={"app": "nonexisting"},
+)
+def test_model_missing_value_in_model(*args):
+    class MyPlugin(PelorusPayload):
```

Comment:
duplicate from line 118. Make it higher level and use in both tests

Reply:
Fixed.